crates/hir_expand/src/db.rs

```diff
@@ -386,11 +386,15 @@ fn expand_proc_macro(db: &dyn AstDatabase, id: MacroCallId) -> ExpandResult<tt::
     };
 
     let attr_arg = match &loc.kind {
-        MacroCallKind::Attr { attr_args, .. } => Some(attr_args),
+        MacroCallKind::Attr { attr_args, .. } => {
+            let mut attr_args = attr_args.0.clone();
+            mbe::Shift::new(&macro_arg.0).shift_all(&mut attr_args);
```

Comment:
This creates a new shift instance on every expansion, is this fine?

If not we can either have the `macro_arg` query return a `Shift` instance which would be only relevant to attribute macros(wasted memory for every other macro I suppose?) or swap things around, shift the attribute item input IDs by the normal attribute input IDs and store the Shift in the `MacroCallKind`.